### PR TITLE
46 Crash when writing cache on invalid state fixed

### LINKED ISSUE #46: Crash when cache is in invalid state even on dispose
Make sure that even if the cache cant get written in the end files will still get cleared from memory and disposed.

### PR DESCRIPTION
**Pull request by frederikja163**
Closes #46

### CODE REVIEW

File: Parquet/ResultListener/ParquetResultListener.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ namespace OpenTap.Plugins.Parquet
     [Display("Parquet", "Save results in a Parquet file", "Database")]
     public sealed class ParquetResultListener : ResultListener
     {
+        public static TraceSource Log { get; } = OpenTap.Log.CreateSource("Parquet");
```
Comment (frederikja163) [inline Parquet/ResultListener/ParquetResultListener.cs:15]:
```suggestion
        internal static TraceSource Log { get; } = OpenTap.Log.CreateSource("Parquet");
```

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
I don't understand how catching an error here will do what is written in issue #46.

Would it be possible to write a unit test that verifies the fix?

**Comment by frederikja163** [comment]
I added a unit test now. The problem is that if we throw an exception while writing the cache the stream and writer will not be disposed. Leaving them to always make errors on subsequent plan runs

**Review by rmadsen-ks — APPROVED** [review]
Looks good!

**Pull request merged**